### PR TITLE
In Android builds, add input_android_physical_keyboard configuration …

### PR DESCRIPTION
**Pull request by gpeche**
…option and its corresponding menu entry to force a device to act as a physical keyboard.

 When running on Android, RetroArch considers most devices that emit dpad events as gamepads, even if they also emit other keyboard events; this is usually the right thing to do, but it has the side effect of not letting some actual keyboards (e.g.: Logitech K480) act as such inside RetroArch. This configuration option allows users to manually select a specific input device to act as a physical keyboard instead of a gamepad, which is handy when emulating computers as opposed to consoles.

## Guidelines

1. Rebase before opening a pull request
2. If you are sending several unrelated fixes or features, use a branch and a separate pull request for each
3. If possible try squashing everything in a single commit. This is particularly beneficial in the case of feature merges since it allows easy bisecting when a problem arises

## Description

[Description of the pull request, detail any issues you are fixing or any features you are implementing]

## Related Issues

[Any issues this pull request may be addressing]

## Related Pull Requests

[Any other PRs from related repositories that might be needed for this pull request to work]

## Reviewers

[If possible @mention all the people that should review your pull request]

### CODE REVIEW

**Comment by LibretroAdmin** [comment]
Hi, all CI jobs are failing with this PR unfortunately. Could you look into fixing that? Thanks.

**Comment by hizzlekizzle** [comment]
Nice, just needed a few more android ifdefs?

This is very welcome functionality, btw. Thanks for the contribution.

**Comment by gpeche** [comment]
> Nice, just needed a few more android ifdefs?
> 
> This is very welcome functionality, btw. Thanks for the contribution.

Yeah, the branch was lingering in my machine for a few days and when I got to make the pull request I botched the last rebase, sorry!

File: configuration.c [hunk 1]
```diff
@@ -4313,7 +4316,7 @@ static void save_keybind_hat(config_file_t *conf, const char *key,
    config[0]        = 'h';
    config[1]        = '\0';
 
-   snprintf(config + 1, sizeof(config) - 1, "%u", hat);
```
Comment (InfamousStarFox) [inline configuration.c:4316]: Looks like a lot of these changes may have been accidentally committed due to a linter adding spaces to the end of lines. To make it easier to read the diff, can you please undo the extraneous spaces which were added?

Comment (gpeche) [reply configuration.c]: Done; also rebased to current master

**Comment by gouchi** [comment]
Reminder : `Settings > Input > Select Physical Keyboard`

image

**Pull request merged**